src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+// Taken from https://github.com/dotnet/cli/blob/8f7b58dd665f60300221a4eb5910a691749396aa/src/Microsoft.DotNet.MSBuildSdkResolver/Interop.NETStandard.cs
+
+#if NETSTANDARD2_0
+using System.Diagnostics;
+
+namespace Microsoft.Build.Locator
+{
+
+    using System;
+    using System.IO;
+    using System.Runtime.InteropServices;
+    using System.Text;
+
+    internal static class DotNetSdkLocationHelper
+    {
+        public static VisualStudioInstance GetInstance(string workingDir)
+        {
+            string dotNetSdkPath = hostfxr_resolve_sdk(workingDir);
+
+            if (String.IsNullOrWhiteSpace(dotNetSdkPath))
+            {
+                return null;
+            }
+
+            if (!File.Exists(Path.Combine(dotNetSdkPath, "Microsoft.Build.dll")))
+            {
+                return null;
+            }
+
+            FileInfo dotnetAssemblyPath = new FileInfo(Path.Combine(dotNetSdkPath, "dotnet.dll"));
+
+            if (!dotnetAssemblyPath.Exists)
+            {
+                return null;
+            }
+
+            FileVersionInfo fileVersionInfo = FileVersionInfo.GetVersionInfo(dotnetAssemblyPath.FullName);
+
+            return new VisualStudioInstance(
+                name: ".NET Core SDK",
+                path: dotNetSdkPath,
+                version: new Version(fileVersionInfo.FileMajorPart, fileVersionInfo.FileMinorPart, fileVersionInfo.FileBuildPart),
+                discoveryType: DiscoveryType.DotNetSdk);
+        }
+
+        private static string hostfxr_resolve_sdk(string workingDir)
+        {
+            Process process = Process.GetCurrentProcess();
```

Comment:
I don't see how `process` can ever be `null`. I don't think that API is capable of returning null so why the check below? That then begs the question can `MainModule` be null, which I don't think that is possible either since I think it will throw on access if it can't be found. Are there cases where `null` is possible?

As a general aside this is my biggest gripe with the `?` operator we now have potential unnecessary `null` checks everywhere due to copy/pasting :-/